hmpy/views/gauge.py

```diff
@@ -0,0 +1,289 @@
+from hmpy.views.view import View
+from PyQt5.QtGui import QPainter, QBrush, QPen, QColor, QImage, QConicalGradient, QPalette, QPixmap, QPolygon, QFont, QFontMetrics
+from PyQt5.QtCore import Qt, QRect, QPoint, QPointF
+from enum import Enum
+import math
+
+
+class GaugeView(View):
+    """
+    View simulating a Gauge.
+    """
+
+    def __init__(self, value=0, min_value=0, max_value=100, precision=2,
+                 unit_text="", color=QColor(Qt.white)):
+        """Initialize the view
+
+        :param starting_value: starting value for the gauge.
+        :param min_value: The minimum value of the gauge.
+        :param max_value: The minimum value of the gauge.
+        :param precision: The minimum value of the gauge.
+        :param unit_text: The text displaying the unit of the gauge
+        :param color: The background colour of the gauge
+        """
+        super().__init__()
+
+        self._color = color
+        self._unit_text = unit_text
+        self.__value = value
+
+        self.__min_value = min_value
+        self.__max_value = max_value
+
+        self._precision = precision
+
+        self._pointText = []
+        self._create_point_text()
+        self._needle_angle = -135
+        self._calculate_needle_angle()
+
+
+    @property
+    def value(self):
+        """Get 'value' class attribute.
+
+        :return: the value of the 'value' attribute.
+        """
+        return self.__value
+
+    @value.setter
+    def value(self, value):
+        """Set 'value' class attribute, trigger repaint.
+
+        :param to: Boolean value.
+        """
+        if value < self.__min_value:
+            self.__value = self.__min_value
+        elif value > self.__max_value:
+            self.__value = self.__max_value
+        else:
+            self.__value = value
+
+        self._calculate_needle_angle()
+        self.repaint()
+
+    @property
+    def min_value(self):
+        """Get 'min_value' class attribute.
+
+        :return: the value of the 'min_value' attribute.
+        """
+        return self.__min_value
+
+    @min_value.setter
+    def min_value(self, min_value):
+        """Set 'min_value' class attribute, trigger repaint.
+
+        :param to: Boolean value.
+        """
+        self.__min_value = min_value
+
+        self._calculate_needle_angle()
+        self.repaint()
+
+    @property
+    def max_value(self):
+        """Get 'max_value' class attribute.
+
+        :return: the value of the 'max_value' attribute.
+        """
+        return self.__max_value
+
+    @max_value.setter
+    def max_value(self, max_value):
+        """Set 'max_value' class attribute, trigger repaint.
+
+        :param to: Boolean value.
+        """
+        self.__max_value = max_value
+
+        self._calculate_needle_angle()
+        self.repaint()
+
+    def paintEvent(self, event):
```

Comment:
There is a lot of seemingly arbitrary values in the paint code.  Without looking to far into it I am wondering if there are any that are worth abstracting into constants or something of that sort? 
I see the number 25 coming up alot, as well as 18, the gradient values, etc.  
This stuff is way beyond what I am familiar with, so I am having trouble really understanding what is going on for some of this.  Just an initial thought

Reply:
Going back and doing the LED view right now, realizing that those values come from my own code. Welp.